pkg/agent/client.go

```diff
@@ -97,12 +97,12 @@ func newAgentClientWithRedialableAgentClient(rac *RedialableAgentClient) *AgentC
 }
 
 // Close closes the underlying stream.
-func (c *AgentClient) Close() {
-	if c.stream == nil {
+func (a *AgentClient) Close() {
```

Comment:
Not something related to this PR, but I think we should fix change the `stream` variable to something more representative to avoid conflicts between grpc stream and network proxy "stream".

- `a.stream` refers to a RedialableAgentClient which contains a grpc stream and conn.
- `a.stream.close()` closes the underlying network proxy stream (the connection), but doesn't affect the grpc stream.

Reply:
Good point. This will be solved if we remove the redialiableAgentClient altogether :)